### DISCUSSION TITLE
Resource not accessible by integration

### DISCUSSION THREAD

**Discussion by saliceti** [bobheadxi/deployments#85] [Help] [answered]
We create a review app with its own environment for each pull request. When the PR is closed, we want the environment to be removed completely.

At the moment we use deactivate-env: https://github.com/DFE-Digital/find-teacher-training/blob/main/.github/workflows/delete-review-app.yml#L85-L91
But the environments are not deleted and I can see dozens in the settings.

We tried with delete-env but it fails with error "Resource not accessible by integration". See: https://github.com/DFE-Digital/find-teacher-training/runs/5317641750?check_suite_focus=true

Any idea?

**Comment by bobheadxi** [comment]
This seems similar to this discussion: https://github.com/bobheadxi/deployments/discussions/78 let me know if none of the approaches described there works for you!

**Comment by saliceti** [comment, marked as answer]
The PAT token is attached to a Github user who may have too many permissions on too many repositories. And it breaks automation when the user leaves. So the only viable solution is to create a dedicated Github service account, which is more work to maintain and potentially more costly. Also it requires admin permission on the repo.
I think the limitation is from Github, but it may be useful to mention it in the Readme?

**Reply by bobheadxi** [reply]
Happy to accept a PR documenting this!
